File: server/src/main/scala/kpn/api/common/monitor/MonitorReferenceInfo.scala
package kpn.api.common.monitor

import kpn.api.common.Bounds
import kpn.api.custom.Timestamp

case class MonitorReferenceInfo(
  created: Timestamp,
  user: String,
  referenceBounds: Bounds,
  referenceDistance: Long,
  referenceType: MonitorReferenceType,
  referenceTimestamp: Timestamp,
  referenceSegmentCount: Long,
  referenceFilename: Option[String]
)

File: backend/api/src/main/java/kpn/api/common/monitor/MonitorReferenceInfo.java
package kpn.api.common.monitor;

import kpn.api.common.Bounds;
import kpn.api.common.monitor.MonitorReferenceType;
import kpn.api.custom.Timestamp;

import java.util.Optional;

public record MonitorReferenceInfo(
  Timestamp created,
  String user,
  Bounds referenceBounds,
  Long referenceDistance,
  MonitorReferenceType referenceType,
  Timestamp referenceTimestamp,
  Long referenceSegmentCount,
  Optional<String> referenceFilename
) {
}
